src-tauri/src/lib.rs

```diff
@@ -294,9 +297,28 @@ pub fn launch_northstar(
     }
 
     // Only Windows with Steam or Origin are supported at the moment
-    if host_os == "windows"
-        && (matches!(game_install.install_type, InstallType::STEAM)
-            || matches!(game_install.install_type, InstallType::ORIGIN)
+    if matches!(game_install.install_type, InstallType::STEAM)
+    {
+        // Create run_northstar.txt for NorthstarProton
+        if host_os != "windows"
+        {
+            let mut file = std::fs::OpenOptions::new()
+                .create(true)
+                .write(true)
+                .truncate(true)
+                .open("run_northstar.txt").unwrap();
+
+            let mut content: &[u8] = b"1";
+            std::io::copy(&mut content, &mut file).unwrap();
```

Comment:
This is copied from `extract`, there are many ways to do file operations in Rust and none of them appear to be that great.

Error handling could be improved to explicitly state why we bailed.

Reply:
Eh if it works it's alright. Wouldn't be the most jank code in the FlightCore code base :P

What I'm wondering more is if we should maybe have a post run call somewhere in FlightCore that cleans up `run_northstar.txt` again if the file wasn't there in the first place?

Otherwise we would have the side-effect of (unintentionally) causing the player to always launch Northstar when they launch Titanfall2 ^^

Reply:
This should 100% be done.

Reply:
Okay, I've been implementing this and...

when do we clean up?
We don't control the launching and just tell Steam to do it.
Do we wait for Titanfall2.exe to launch and then move the files back?
Or maybe for Titanfall2.exe to exit? but then we need to persist after the games launch.

Reply:
On a sidenote, given that control is starting to get messy (and needs restructuring anyway, looking at #179), it might make sense to just move all the code for Steam launching into a dedicated function and have it hooked up to a button in DeveloperView.

Then we can just ship that in an update and have people test it that way.

Then we refactor the launching process and from there incorporate the function for launching via Steam. ^^

Reply:
> I'd change it back after `NorthstarLauncher.exe` was detected as running.

We don't run `NorthstarLauncher.exe` on Linux (or Windows when launching over Steam).
We cannot replace `Titanfall2.exe` because that would make the install unclean possible causing future updates to reset it.
The launch parameters cannot be hijacked to launch `NorthstarLauncher.exe` either because its setup to interact with EA's App directly (see [SteamDB](https://steamdb.info/app/1237970/config/))

So there only really is `Titanfall2.exe`
There might be some way of checking if the running Titanfall process is Northstar by seeing what is loaded, but I think that is a bit overkill.
It is fair to assume any Titanfall2.exe or NorthstarLauncher.exe might be Northstar, in which case launching should be prevented.

> Cause at that point the wsock32 proxy did its job and launched Northstar.

unsure.
wsock32 is a runtime dependency and there is no reliable duration it will take for it to load the configuration as it can depend on multiple factors (e.g. computer hardware)

Inotify can detect when a file was opened, that could be used to check when Northstar read the file, its possibly something similar exists for Windows

> Only issue is when user closes FlightCore before Northstar launched. Don't really have a solution for that one tbh...

While I agree that it would be best to clean up after ourselves and bring the environment back into the state we found it in, maybe there is no good way of doing this.

Reply:
> We don't run `NorthstarLauncher.exe` on Linux (or Windows when launching over Steam).

Oh I thought it starts it as a new process nevermind then my mistake ^^

Will require some testing but it might be enough to check for a running `Titanfall2.exe` then and after X seconds start the clean-up process.

Reply:
Think this is resolved by now ^^